### PR TITLE
550 associating saved searches and lists with a job should done by selecting the job instead of pasting salesforce link

### LINKED ISSUE #550: Employer access: Associating saved Searches and Lists with a job should done by selecting the job instead of pasting Salesforce link
As an employer
I can't use Salesforce links to identify jobs
because I don't have access to Salesforce

Similarly instead of a Salesforce link to an Account, I should be able to be able default to myself

### PR DESCRIPTION
**Pull request by camerojo**

### CODE REVIEW

File: ui/admin-portal/src/app/components/util/joblink/joblink.component.ts [hunk 1]
```diff
@@ -14,107 +14,118 @@
...
+  ngOnChanges(changes: SimpleChanges): void {
+    //Display appropriate heading for job search
+    this.searchHeading = (this.jobId ? "Change" : "Optional") + " job association";
 
-      return retval;
-    };
+    //If there is already a job associated, get its name and construct the default
+    //job request (ie to retain existing job)
+    if (this.jobId) {
+      this.jobService.get(this.jobId).subscribe({
```
Comment (sadatmalik) [inline ui/admin-portal/src/app/components/util/joblink/joblink.component.ts:94]: Good to unsubscribe this when the component is destroyed.

```
private jobSubscription: Subscription;

ngOnChanges(changes: SimpleChanges): void {

  . . .

  // ngOnChanges may be called many times, so unsubscribe from the previous subscription if it exists?
  if (this.jobSubscription) {
    this.jobSubscription.unsubscribe();
  }

  this.jobSubscription = this.jobService.get(this.jobId).subscribe({
    // subscription logic
  });
}


ngOnDestroy(): void {
  // Ensure all subscriptions are unsubscribed when the component is destroyed
  if (this.jobSubscription) {
    this.jobSubscription.unsubscribe();
  }

```

**Review by sadatmalik — COMMENTED** [review]
Left one comment in JoblinkComponent

File: ui/admin-portal/src/app/components/util/joblink/joblink.component.ts (hunk 1 above)
Comment (camerojo) [reply ui/admin-portal/src/app/components/util/joblink/joblink.component.ts:94]: Hi @sadatmalik I know it is a little surprising, but the job service subscription is to an Http Observable - and you don't need to unsubscribe from them. See https://angular.io/guide/http-request-data-from-server#:~:text=In%20general%2C%20you%20should%20unsubscribe,request%20responds%20or%20times%20out.

Comment (sadatmalik) [reply ui/admin-portal/src/app/components/util/joblink/joblink.component.ts:94]: Thanks for clarifying that @camerojo - approved for merging.

**Review by sadatmalik — DISMISSED** [review]
Approved

**Pull request merged**